contracts/LicenseToken.sol

```diff
@@ -0,0 +1,333 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.23;
+import { Base64 } from "@openzeppelin/contracts/utils/Base64.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+// solhint-disable-next-line max-line-length
+import { ERC721EnumerableUpgradeable, ERC721Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721EnumerableUpgradeable.sol";
+import { IERC721Metadata } from "@openzeppelin/contracts/token/ERC721/extensions/IERC721Metadata.sol";
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import { ILicenseToken } from "./interfaces/ILicenseToken.sol";
+import { ILicensingModule } from "./interfaces/modules/licensing/ILicensingModule.sol";
+import { IDisputeModule } from "./interfaces/modules/dispute/IDisputeModule.sol";
+import { Errors } from "./lib/Errors.sol";
+import { GovernableUpgradeable } from "./governance/GovernableUpgradeable.sol";
+import { ILicenseTemplate } from "./interfaces/modules/licensing/ILicenseTemplate.sol";
+
+/// @title LicenseToken aka LNFT
+contract LicenseToken is ILicenseToken, ERC721EnumerableUpgradeable, GovernableUpgradeable, UUPSUpgradeable {
```

Comment:
FYI, since we are using ERC721Enumerable, it might be worth revisiting ERC721A which is significantly cheaper for batch mints than Enumerable. 

When minting 5 LTs at once, 500k in gas saving (of A vs. Enumerable) is worth at least 10 cents on L2s like Base.

Reply:
we don't need Enumerable AFAIK, we used to go with just ERC721

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/ERC721.sol

ERC721Enumerable is an extension:
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/extensions/ERC721Enumerable.sol

The gas hit is keeping this (allows to decouple id from order of creation)
```
uint256[] private _allTokens;
mapping(uint256 tokenId => uint256) private _allTokensIndex;
```
We are not using tokenByIndex() and we use sequential ids, so we can just use ERC721 and use a uint counter for the ids.

Reply:
As discussed offline, since the license tokens are expected to be frequently burnt and the token IDs will not be consecutive, the enumerable API will unblock crucial onchain use cases. Although gas costs are higher compared to normal ERC721, if we decide to change to normal ERC721 in the future, the upgrade path is also relatively easier.